Controller/ContentManagement/CrudController.php

```diff
@@ -358,4 +374,47 @@ public function getContentTypeInfo(ContentType $contentType)
                 'contentType' => $contentType,
         ]);
     }
+
+    /**
+     * @Route("/api/user-profile", defaults={"_format": "json"}, methods={"GET"})
+     */
+    public function getUserProfile() : JsonResponse
+    {
+        $user = $this->getUser();
+        if (! $user instanceof User) {
+            throw new \RuntimeException('User profile class not recognized');
+        }
+        if (! $user->isEnabled()) {
+            throw new \RuntimeException('User disabled');
+        }
+
+        return $this->json([
+            'username' => $user->getUsername(),
+            'displayName' => $user->getDisplayName(),
+            'roles' => $user->getRoles(),
+            'email' => $user->getEmail(),
+            'circles' => $user->getCircles(),
+        ]);
+    }
+
+    /**
+     * @Route("/api/user-profiles", defaults={"_format": "json"}, methods={"GET"})
+     * @IsGranted("ROLE_ADMIN")
+     */
+    public function getUserProfiles() : JsonResponse
+    {
+        $users = [];
+        foreach ($this->userService->getAllUsers() as $user) {
+            if ($user->isEnabled()) {
+                $users[] = [
+                    'username' => $user->getUsername(),
+                    'displayName' => $user->getDisplayName(),
+                    'roles' => $user->getRoles(),
+                    'email' => $user->getEmail(),
+                    'circles' => $user->getCircles(),
+                ];
+            }
+        }
+        return $this->json($users);
```

Comment:
same comment, the controller should be the glue.

```suggestion
        return $this->json($this->userService->getEnabledUsers());
```

Reply:
To me the $this->userService->getEnabledUsers() could expose too much info, we should have something like $this->apiService->getUsers()? What do you think?

Reply:
I do not get why $this->userService->getEnabledUsers() would expose too much information?
You are allready exposing this info in the controller (and the controller should not contain this logic). The service is part of your backend, why are you worried?

Reply:
See 9cbf75b7315e4fec4fa3edfeea82918a61565542

Reply:
even better in the user entity fda1219a36c1bec8f610426fdb8efc358057a658